src/averageAges.js

```diff
@@ -55,9 +65,50 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const peopleWithMother = people.filter(person => people
+    .some(mother => (
+      mother.name === person.mother && (
+        onlyWithSon
+          ? person.sex === 'm'
+          : true
+      )
+    ))
+  );
+
+  const sumOfDifferences = peopleWithMother
+    .map((person) => {
```

Comment:
You don't need both the `map` and `reduce` methods. Call only `reduce` and do all calculations in it.